presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/Rule.java

```diff
@@ -46,4 +57,9 @@ default boolean isEnabled(Session session)
 
         Session getSession();
     }
+
+    interface Result
+    {
+        Result transformTo(PlanNode planNode);
```

Comment:
`transformTo` suggest that the rule wants to replace root node with some other equivalent node. While this is the case for some rules (e.g: Unalias/Prune since there is no point in keeping the original node), other (majority?) rules would just add equivalent plan nodes so that all of them are cost evaluated. What about such methods:
```
// adds a plan node equivalent to rule execution root plan node 
addPlanNode(PlanNode node) 
// replaces rule execution root plan node
replacePlanNode(PlanNode node) 
```
It is more explicit. `IterativeOptimizer` would just keep the argument of the last call to `add/replacePlanNode`, so that same rules can be used (ported) in iterative/equivalence optimizer.
What do you think?

FYI: @martint ?